### DISCUSSION TITLE
Support for macros returning macros?

### DISCUSSION THREAD

**Discussion by kiranandcode** [racket/rhombus#298] [Q&A] [answered]
Sorry for the multiple questions. This follows on from @mflatt 's earlier response to supporting `+=` operators.

Is it possible to write a macro that automatically generates `dot` definitions for a class?

Here's what I've tried so far:

```rhombus
defn.macro 'object_with_fields $name (($field, $get_field, $set_field), ...)':
  'class $name($field,...):
    opaque
    internal $_name

    constructor ($field, ...):
      super($field, ...)

    dot $(@{'}) $('$') left . $field $(@{'}) :
      ~tail: tail
      match tail                        //other cases omitted for now
      | ~else:
         $(@{'})block:
           def b :~ $_name =  @{$} left
           b . $field
         $(@{'})
    ...
    '
```
The above fails with the error `dot: bad syntax`, which I assume is because `${@{'})` isn't actually what I want, and the code I'm emitting isn't right: 
```c
// dot: bad syntax
//  in: (dot "'" (op $) left (op |.|) x "'" (block ,,,
```

**Comment by mflatt** [comment, marked as answer]
The trick to nesting quotes in quotes is to use `'«` for the outer opening quote and `»'` for the outer closing quote:

```
defn.macro 'object_with_fields ....':
  '«class $name($field,...):
      ....
      dot '$('$') left . $field':
        ....
      ...
      »'
```

You can use `'«`...`»'` in place of `'`...`'` anywhere, but the idea of using non-ASCII characters is that immediate quote nesting is relatively rare, so maybe it's ok to make that a little more difficult to type (but `«` and `»` are Option-\ and Shift-Option-\ on a U.S. Mac keyboard), and that way we don't use more valuable characters and combinations.
